File: server/model/system/request/casbin.go
package request

type CasbinInfo struct {
	Path   string `json:"path"`   // 路径
	Method string `json:"method"` // 方法
}

type ReqCasbin struct {
	RoleId      uint         `json:"roleId" validate:"required"`
	CasbinInfos []CasbinInfo `json:"casbinInfos"`
}

func DefaultCasbin() []CasbinInfo {
	return []CasbinInfo{
		{Path: "/base/captcha", Method: "POST"},
		{Path: "/base/login", Method: "POST"},
		{Path: "/user/getUserInfo", Method: "GET"},
		{Path: "/menu/getMenus", Method: "GET"},
		{Path: "/user/editUser", Method: "POST"},
		{Path: "/user/modifyPass", Method: "POST"},
		{Path: "/jwt/joinInBlacklist", Method: "POST"},
	}
}

File: web/src/common/apis/base/casbin.ts
import { request } from "@/http/axios"

export interface CasbinInfo {
  path: string
  method: string
}

interface reqCasbin {
  roleId: number
  casbinInfos: CasbinInfo[]
}

export function editCasbinApi(data: reqCasbin) {
  return request<ApiResponseData<null>>({
    url: "/casbin/editCasbin",
    method: "post",
    data
  })
}
